### DISCUSSION TITLE
(ANSWERED) NEW almanac FUNCTION: planet entry zodiac sign

### DISCUSSION THREAD

**Discussion by gipsea** [skyfielders/python-skyfield#609] [Ideas]
Dear all,
I'm not sure if this function will help someone but I'd like to contribute to skyfield if I can with what I'm producing.

Below you can find the function I've prepared to study the entry of planets into a zodiac sign either in Heliocentric then geocentric situation.

```
ENTRY_ZODIAC = ['Aries','Taurus','Gemini',
          'Cancer','Leo','Virgo',
          'Libra','Scorpio','Sagittarius',
          'Capricorn','Aquarius','Pisces']

def entry_zodiac(ephemeris, base, target):
    """Buid a function which return date and time when a planet enter
    a zodiac sign. It works in heliocentric and geocentric
    base: sun or Earth
    Target: planet to study
    """
    if base == ephemeris['sun']:
        sun_at = base.at
        def entry_zodiac_at(t):
            """Return the Zodiac sign 0 through 113 at time `t`."""
            s = sun_at(t)
            _, tlon, _ = s.observe(target).ecliptic_latlon()
            return (tlon.radians // (tau / 12) % 12).astype(int)
    else:
        earth_at = base.at
        def entry_zodiac_at(t):
            """Return the Zodiac sign 0 through 113 at time `t`."""
            e = earth_at(t)
            _, tlon, _ = e.observe(target).apparent().ecliptic_latlon()
            return (tlon.radians // (tau / 12) % 12).astype(int)
    entry_zodiac_at.step_days = 1.0  # I use 1.0 becaue I move from quick and fast planets
    return entry_zodiac_at
```

cheers

ps: updated the post as suggested for better reading of the code

**Comment by gipsea** [comment]
sorry...I-m not too familiar with editing code in github :)

**Reply by brandon-rhodes** [reply]
Try putting a line with three back-tick characters at the top and bottom of the code; that should fix the formatting so it's easier to read.

Since Skyfield has an `almanac_east_asia.py` module that computes the solar term (24 around the sky) of a position, I suppose it would make sense for there to be an `almanac_europe.py` (would that be the right name, I wonder? `_greek.py` might be a little too specific) that does houses instead (12 around the sky). I'm open to a pull request offering such a module. In the meantime, thanks for sharing the code, as folks can paste it into their own projects!

**Reply by gipsea** [reply]
Ciao Brandon,
I'll take a look at the almanac_east_asia.py asap and let you know if there is any code I'll need before you we start some changes.

By all mean at the moment I'm considering all computation @ 00:00UTC and _europe it is probably good enough;)
cheers
